app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php

```diff
@@ -27,6 +40,66 @@ class AddComment extends \Magento\Sales\Controller\Adminhtml\Order implements Ht
      */
     const ADMIN_SALES_EMAIL_RESOURCE = 'Magento_Sales::emails';
 
+    /**
+     * @var OrderCommentSender
+     */
+    private $orderCommentSender;
+
+    /**
+     * @var GridPool
+     */
+    private $gridPool;
+
+    /**
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param FileFactory $fileFactory
+     * @param InlineInterface $translateInline
+     * @param PageFactory $resultPageFactory
+     * @param JsonFactory $resultJsonFactory
+     * @param LayoutFactory $resultLayoutFactory
+     * @param RawFactory $resultRawFactory
+     * @param OrderManagementInterface $orderManagement
+     * @param OrderRepositoryInterface $orderRepository
+     * @param LoggerInterface $logger
+     * @param OrderCommentSender $orderCommentSender
+     * @param GridPool $gridPool
+     *
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+     */
+    public function __construct(
+        Context $context,
+        Registry $coreRegistry,
+        FileFactory $fileFactory,
+        InlineInterface $translateInline,
+        PageFactory $resultPageFactory,
+        JsonFactory $resultJsonFactory,
+        LayoutFactory $resultLayoutFactory,
+        RawFactory $resultRawFactory,
+        OrderManagementInterface $orderManagement,
+        OrderRepositoryInterface $orderRepository,
+        LoggerInterface $logger,
+        ?OrderCommentSender $orderCommentSender = null,
+        ?GridPool $gridPool = null
+    ) {
+        $this->orderCommentSender = $orderCommentSender ?? ObjectManager::getInstance()->get(OrderCommentSender::class);
+        $this->gridPool = $gridPool ?? ObjectManager::getInstance()->get(GridPool::class);
```

Comment:
I don't think adding these entries as null as default and solving it with ObjectManager would be the best approach here.

I think that won't be anything broken simply adding those classes. 

If it breaks anything I would go deeper and update the broken files. It shouldn't be long.